packages/node_modules/@ciscospark/widget-recents/src/selector.js

```diff
@@ -141,16 +143,18 @@ function sortByNewest(space) {
 }
 
 export const getRecentsWidgetProps = createSelector(
-  [getWidget, getUsers, getSpaces, getRecentSpacesWithAvatarUrl, getSpark, getMedia, getTeams],
-  (widget, user, spaces, spacesList, spark, media, teams) => {
+  [getWidget, getUsers, getSpaces, getRecentSpacesWithAvatarUrl, getSpark, getMedia, getTeams, getFeatures],
+  (widget, user, spaces, spacesList, spark, media, teams, features) => {
     let lastActivityDate;
     if (spacesList && spacesList.count()) {
       lastActivityDate = spacesList.last().lastActivityTimestamp;
     }
+    const hasGroupCalling = features.getIn([`items`, FEATURE_GROUP_CALLING]);
     return { // eslint-disable-line max-params
-      widgetRecent: widget,
+      widgetStatus: widget.get(`status`).toJS(),
```

Comment:
Good call on being more specific here. Does the `toJS()` force too many rerenders or is the shallow comparison working? Would it be more performant if we convert the `getWidget` function to directly grab `state.widgetRecents.get('status')`?

Reply:
I honestly don't know. Would love to get these selectors as performant as possible, but not certain the best approach. Right now, as a starting step, I'm just trying to remove the immutable conversion from the renderer and into the selector, hopefully allowing the memoization to help.